gitlab/client.py

```diff
@@ -98,7 +109,9 @@ def __init__(
         self._set_auth_info()
 
         #: Create a session object for requests
-        self.session = session or requests.Session()
+        http_engine = kwargs.pop("http_engine", http_engines.DefaultEngine)
+        self.http_engine = http_engine(**kwargs)
+        self.session = self.http_engine.client
```

Comment:
note to self, Most likely we will want to get rid of this `self.session` and abstract it in the engine/backend code so each implementation implements `close()` internally But we can leave it here for now.